src/printer.js

```diff
@@ -1209,13 +1209,17 @@ function genericPrintNoParens(path, options, print) {
       else parts.push("default:");
 
       if (n.consequent.find(node => node.type !== "EmptyStatement")) {
-        const cons = path.call(
-          function(consequentPath) {
-            return printStatementSequence(consequentPath, options, print);
-          },
-          "consequent"
-        );
-
+        const parent = path.getParentNode();
+        const lastCase = util.getLast(parent.cases);
+        const cons = path.call(consequentPath => {
+          let printed = [];
+          path.map(p => {
```

Comment:
I just realized that map creates an array with the returned values. So you can do

```js
return join(hardline, path.map(p => {
  ...
  return concat(...);
});
```

to make the code shorter.

Sorry my feedback is coming in waves. This is the only last thing I'm seeing with your code!